### DISCUSSION TITLE
Additional push in a Dependabot PR does not trigger actions

### DISCUSSION THREAD

**Discussion by nak3** [community/community#62346] [Code Security] [answered]
### Select Topic Area

Bug

### Body

Our repositories have some code generation scripts that need to run when updating dependencies.
So we setup a script to run after dependabot to bump these dependencies, then push the change in the PR.

The code is  (It is based on the discussion https://github.com/orgs/community/discussions/48498):

```
jobs:
  generate_some_code:
    name: update deps
    runs-on: ubuntu-latest
    if: ${{ github.actor == 'dependabot[bot]' }}
    steps:
      - name: Checkout
        uses: actions/checkout@v3
        with:
          ref: ${{ github.head_ref }}
          path: ./src/github.com/${{ github.repository }}
          fetch-depth: 0

      - name: Run hack scripts
        working-directory: ./src/github.com/${{ github.repository }}
        run: |
          ./hack/update-deps.sh
          ./hack/update-codegen.sh

      - name: git push
        working-directory: ./src/github.com/${{ github.repository }}
        run: |
          if ! git diff --exit-code --quiet
          then
            git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
            git config --local user.name "github-actions[bot]"
            git add .
            git commit -m "Run ./hack/update-codegen.sh"
            git push
          fi
```

However, any github action is not triggered when the script push to the change.
So far, I have observed the following conditions:

- The `Checks` tab displays `There are no checks for this commit`.
- The github action is NOT triggered when the above script pushed.
- Tried to call `@dependabot {rebase, recreate}` after the script pushed but it does NOT work.
- Closing & re-opening the PR (remove&restore the branch is not necessary) after the script pushed but does work.
- The github action was sometimes triggered a few weeks ago -  Jul 11, 2023. (Note, but it is different repo so it might have some diff.) 
e.g. https://github.com/knative/pkg/pull/2775

**Comment by krsna-m** [comment, marked as answer]
Opened an issue and this was the reply from github:

> Please note that this is the expected and documented behavior:
> 
> https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow
> 
> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.
> 
> The actions/checkout action uses the GITHUB_TOKEN to checkout the repository and these credentials are cached, and will be used in subsequent steps (when making the push). You'll need to use a [Personal Access Token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) or a [GitHub App token](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/managing-deploy-keys#github-app-installation-access-tokens) when making that push if you expect GitHub Actions workflows to be triggered as a result.

**Reply by nak3** [reply]
Thank you so much @kvmware !
